libsignal-service/src/sender.rs

```diff
@@ -680,53 +692,57 @@ where
         &mut self,
         recipient: &ServiceAddress,
         unidentified_access: Option<&UnidentifiedAccess>,
-        device_id: i32,
+        device_id: u32,
         content: &[u8],
     ) -> Result<OutgoingPushMessage, MessageSenderError> {
         let recipient_address = get_preferred_protocol_address(
-            &self.cipher.store_context,
-            recipient.clone(),
+            &mut self.session_store,
+            recipient,
             device_id,
-        )?;
+        )
+        .await?;
         log::trace!("encrypting message for {:?}", recipient_address);
 
         if !self
-            .cipher
-            .store_context
-            .contains_session(&recipient_address)?
+            .session_store
+            .load_session(&recipient_address, None)
+            .await?
+            .is_some()
```

Comment:
Suggestion feature on GH doesn't let me, but this should be `self.[..].is_none()` instead, apparently